src/main/groovy/org/labkey/gradle/task/CreateModule.groovy

```diff
@@ -0,0 +1,169 @@
+package org.labkey.gradle.task
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.CopySpec
+import org.gradle.api.tasks.TaskAction
+
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+/**
+ * Creates a new module based on the contents of trunk/server/moduleTemplate. Sources properties from commandline or
+ * user input based on the prompt flag.
+ */
+class CreateModule extends DefaultTask
+{
+    boolean prompt = true
+
+    @TaskAction
+    void createModule() {
+        String moduleName = null
+        String moduleDestination = null
+        boolean hasManagedSchema = true
+        boolean createTestFiles = true
+        boolean createApiFiles = true
+
+        if (prompt) {
+            project.ant.input(
+                    message: "\nEnter the name for your new module: ",
+                    addProperty: "new_moduleName"
+            )
+            moduleName = ant.new_moduleName
+
+            project.ant.input(
+                    message: "\nEnter the full path for where to put the new module: ",
+                    addProperty: "new_moduleDestination"
+            )
+            moduleDestination = ant.new_moduleDestination
+
+            project.ant.input(
+                    message: "\nWill this module create and manage a database schema? (y/N)",
+                    addProperty: "new_hasManagedSchema"
+            )
+            hasManagedSchema = !(ant.new_hasManagedSchema.toLowerCase().equals("n"))
+
+            project.ant.input(
+                    message: "\nCreate test stubs (y/N)",
+                    addProperty: "new_createTestFiles"
+            )
+            createTestFiles = ant.new_createTestFiles.toLowerCase().equals("y")
+
+            project.ant.input(
+                    message: "\nCreate API stubs (y/N)",
+                    addProperty: "new_createApiFiles"
+            )
+            createApiFiles = ant.new_createApiFiles.toLowerCase().equals("y")
+        }
+        else {
+            if (project.hasProperty('moduleName')) {
+                moduleName = project.moduleName
+            }
+            if (project.hasProperty('moduleDestination')) {
+                moduleDestination = project.moduleDestination
+            }
+            if (project.hasProperty('hasManagedSchema')) {
+                hasManagedSchema = !(project.hasManagedSchema.toLowerCase().equals("n"))
+            }
+            if (project.hasProperty('createTestFiles')) {
+                createTestFiles = project.createTestFiles.toLowerCase().equals("y")
+            }
+            if (project.hasProperty('createApiFiles')) {
+                createApiFiles = project.createApiFiles.toLowerCase().equals("y")
```

Comment:
For these boolean properties, let's not require a value.  Instead, just check for the existence of the property.  Then someone can use them on the command line like so:
-PcreateTestFiles -PnoManagedSchema